modules/auxiliary/gather/ldap_query.rb

```diff
@@ -67,75 +67,75 @@ def initialize(info = {})
+      template = File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'ldap_query', 'ldap_queries_template.yaml')
+      FileUtils.cp(template, user_config_file_path) if File.exist?(template)
     end
 
     # Combine the user settings with the default settings and then uniq them such that we only have one copy
     # of each ACTION, however we use the user's custom settings if they have tweaked anything to prevent overriding
     # their custom adjustments.
-    @default_settings['queries'].concat(user_settings['queries'])
-    @default_settings['queries'].uniq! do |q|
-      q['action']
-    end
+    @loaded_queries = Hash[*@loaded_queries.map { |h| [h['action'], h] }.flatten]
+    @loaded_queries.select! do |_, entry|
+      if entry['action'].blank?
+        wlog('ldap query entry detected that was missing its action field')
+        return false
+      end
 
-    actions = []
-    @default_settings['queries'].each do |entry|
-      if entry['action'].nil? || entry['description'].nil?
-        if entry['action'].nil?
-          print_warning("Entry detected that was missing its 'action' and 'description' fields!")
-        else
-          print_warning("#{entry['Action']} is missing its 'description' field!")
-        end
-        print_warning("Please check the files at #{@default_settings_file_path} and #{@user_config_file_path} and fix the errors listed above!")
-        next
+      if %w[RUN_QUERY_FILE RUN_SINGLE_QUERY].include? entry['action']
+        wlog("ldap query entry detected that was using a reserved action name: #{entry['action']}")
+        return false
       end
-      next if entry['action'] == 'SAMPLE_ACTION' # Skip the sample action in the file.
 
-      actions << [entry['action'], { 'Description' => entry['description'] }]
+      if entry['filter'].blank?
+        wlog('ldap query entry detected that was missing its filter field')
+        return false
+      end
+
+      unless entry['attributes'].is_a? Array
+        wlog('ldap query entry detected that was missing its attributes field')
+        return false
+      end
+
+      true
+    end
+
+    actions = []
+    @loaded_queries.each_value do |entry|
+      actions << [entry['action'], { 'Description' => entry['description'] || '' }]
     end
     actions << ['RUN_QUERY_FILE', { 'Description' => 'Execute a custom set of LDAP queries from the JSON or YAML file specified by QUERY_FILE.' }]
     actions << ['RUN_SINGLE_QUERY', { 'Description' => 'Execute a single LDAP query using the QUERY_FILTER and QUERY_ATTRIBUTES options.' }]
     actions.sort!
 
     default_action = 'RUN_QUERY_FILE'
-    if actions.length > 2 # Aka there is more than just RUN_QUERY_FILE and RUN_SINGLE_QUERY in the list...
+    unless @loaded_queries.empty? # Aka there is more than just RUN_QUERY_FILE and RUN_SINGLE_QUERY in the list...
```

Comment:
Should probably update this comment a bit. Its still relevant but needs to be worded slightly differently.